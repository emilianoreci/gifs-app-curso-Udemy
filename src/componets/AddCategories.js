//  @Package
import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategories = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    //console.log("handleSubmit--llamado, ", inputValue); para el test

    //En vez de recibir las categories por props
    //setCategories([...categories, inputValue]);
    if (inputValue.trim().length > 2) {
      setCategories((cate) => [inputValue, ...cate]); //es un set de useState()
      setInputValue("");
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <h2>AddCategories</h2>
      <p>{inputValue}</p>
      <input type="text" value={inputValue} onChange={handleInputChange} />
    </form>
  );
};

AddCategories.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
