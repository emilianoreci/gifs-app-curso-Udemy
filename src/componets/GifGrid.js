//  @Package
import React from "react";
import { PropTypes } from "prop-types";

//  @Own
import { GifGridItem } from "./GifGridItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category }) => {
  //hago destructuring y le cambio el nombre de data a images para usarlo.
  const { data: images, loading } = useFetchGifs(category);
  return (
    <>
      <h2 className="animate__animated animate__fadeIn">{category}</h2>
      <div className="card-grid">
        {loading && <p>cargando...</p>}
        {images.map((eachElement) => (
          <GifGridItem key={eachElement.id} {...eachElement}></GifGridItem>
          /* <GifGridItem key={eachElement.id} img={eachElement}></GifGridItem> */
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
