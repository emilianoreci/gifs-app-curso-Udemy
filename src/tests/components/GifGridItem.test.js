import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { GifGridItem } from "../../componets/GifGridItem";

describe("Pruebas al <GifGridItem />", () => {
  const title = "La musica de hoy";
  const url = "https://localhost/algo";

  let wrapper = shallow(<GifGridItem title={title} url={url} />);

  beforeEach(() => {
    wrapper = shallow(<GifGridItem title={title} url={url} />);
  });

  test("Debe mostrar correctamente el componente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe tener un <p/> con el title", () => {
    expect(wrapper.find("p").text().trim()).toBe(title);
  });

  test("La etiqueta img debe recibir los atributos/props src y alt", () => {
    const img = wrapper.find("img").props(); //prop() singular o props() plural
    //console.log(img.props().src);
    //console.log(img.props());
    //<img src="https://localhost/algo" alt="La musica de hoy"/>

    expect(img.src).toBe(url);
    expect(img.alt).toBe(title);
  });

  test("El div debe de tener la clase animate__fadeIn ", () => {
    const div = wrapper.find("div");
    const className = div.prop("className");

    expect(className.includes("animate__fadeIn")).toBe(true);
  });
});
