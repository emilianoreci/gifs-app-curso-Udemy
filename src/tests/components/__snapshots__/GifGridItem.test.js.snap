// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al <GifGridItem /> Debe mostrar correctamente el componente 1`] = `
<div
  className="card animate__animated animate__fadeIn "
>
  <img
    alt="La musica de hoy"
    src="https://localhost/algo"
  />
  <p>
    La musica de hoy
  </p>
</div>
`;
